### DISCUSSION TITLE
Multipath devices not detected as OSDs

### DISCUSSION THREAD

**Discussion by weeix** [rook/rook#11787] [Q&A] [answered]
Hello there, I'm attempting to add multipath devices as OSDs. I already tried:

- `deviceFilter: ^mpath.`
- `deviceFilter: ^dm-[0-9]+`
- `devicePathFilter: ^/dev/mapper/mpath.`
- `devicePathFilter: ^/dev/disk/by-id/dm-uuid-mpath-.*`
- ```
  devices:
  - name: /dev/disk/by-id/dm-uuid-mpath-<REDACTED>
  ```

But none of the aforementioned settings actually work. Has anyone used multipath devices as OSDs with success?

In my `values.yaml` (HelmChart), I also included `logLevel: DEBUG`, and used `kubectl -n rook-ceph logs rook-ceph-operator-<REDACTED> | grep '/dev` to see the logs, but were unable to find anything besides the lines stating the changes after I used `kubectl -n rook-ceph edit cephcluster rook-ceph`

**Comment by microyahoo** [comment, marked as answer]
Each node has own `rook-ceph-osd-prepare` pod to configure osds, so you can check coresponding pod log to identify the cause.

**Reply by weeix** [reply]
Thanks, I'll try to check the logs tomorrow and report back.

**Reply by weeix** [reply]
Viewing the logs was very helpful. I came across an entry that stated `I | cephosd: skipping device "dm-2": ["Has GPT headers"].` and used `sudo wipefs -a /dev/mapper/mpatha` command to zap the GPT headers. I can now use `devicePathFilter: ^/dev/disk/by-id/dm-uuid-mpath-.*` to include the multipath devices with success.

**Comment by richardelling** [comment]
An alternative approach that we use for complex or unpredictable fabric configurations is to use PVCs. So we have direct (imperative) control of how the PV is configured and rook only uses the drives via PVCs.

**Reply by weeix** [reply]
Thank you. PVCs based cluster is always a mysterious topic for me. I firstly thought that it's intended for public clouds, but then I saw the local storage PVC example in the doc, and always wonder about the use cases. So this is one of it.
